hack/release/common.sh

```diff
@@ -0,0 +1,163 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+config(){
+  GITHUB_ACCOUNT="Azure"
+  RELEASE_ACR=${RELEASE_ACR:-ksnap.azurecr.io} # will always be ovreridden
+  RELEASE_REPO_ACR=${RELEASE_REPO_ACR:-${RELEASE_ACR}/public/aks/karpenter/}
+  RELEASE_REPO_MAR=mcr.microsoft.com/aks/karpenter
+  SNAPSHOT_ACR=${SNAPSHOT_ACR:-ksnap.azurecr.io}
+  SNAPSHOT_REPO_ACR=${SNAPSHOT_REPO_ACR:-${SNAPSHOT_ACR}/karpenter/snapshot/}
+
+  CURRENT_MAJOR_VERSION="0"
+  RELEASE_PLATFORM="--platform=linux/amd64,linux/arm64"
+
+  MAIN_GITHUB_ACCOUNT="Azure"
+  RELEASE_TYPE_STABLE="stable"
+  RELEASE_TYPE_SNAPSHOT="snapshot"
+}
+
+# versionData sets all the version properties for the passed release version. It sets the values
+# RELEASE_VERSION_MAJOR, RELEASE_VERSION_MINOR, and RELEASE_VERSION_PATCH to be used by other scripts
+versionData(){
+  local VERSION="$1"
+  local VERSION="${VERSION#[vV]}"
+  RELEASE_VERSION_MAJOR="${VERSION%%\.*}"
+  RELEASE_VERSION_MINOR="${VERSION#*.}"
+  RELEASE_VERSION_MINOR="${RELEASE_VERSION_MINOR%.*}"
+  RELEASE_VERSION_PATCH="${VERSION##*.}"
+  RELEASE_MINOR_VERSION="v${RELEASE_VERSION_MAJOR}.${RELEASE_VERSION_MINOR}"
+}
+
+snapshot() {
```

Comment:
nit: should this live in `snapshot.sh`?

Reply:
It could. Though this follows some patterns established by the AWS provider that I like. The general idea is that snaphshot/release.sh are just thin entry points, most of the actual logic for both is in common.sh. Among other things, this makes it easy to see the differences and commonalities, update them in tandem, etc. I would prefer to keep it this way.

Reply:
Sure, that's fine. Don't have a strong preference. I am mostly familiar with the common element containing only what's needed.